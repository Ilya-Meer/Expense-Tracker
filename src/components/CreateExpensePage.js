import React, { Component } from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { H2 } from './StyledComponents/Shared';
import { startAddExpense } from '../actions/expenseActions';

class CreateExpensePage extends Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onSubmit(expense) {    
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <H2>Add an Expense</H2>
        <ExpenseForm onSubmit={this.onSubmit}/>
      </div>    
    )
  }
}

export { CreateExpensePage }


const mapDispatchToProps = (dispatch) => {
  return {
    startAddExpense: (expense) => {
      dispatch(startAddExpense(expense))
    }
  }
}
export default connect(undefined, mapDispatchToProps )(CreateExpensePage);