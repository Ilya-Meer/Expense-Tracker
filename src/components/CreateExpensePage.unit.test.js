import React from 'react';
import { shallow } from 'enzyme';
import {CreateExpensePage} from './CreateExpensePage';
import moment from './__mocks__/moment'

describe('Create Expense Page', () => {
  let onSubmit, history, wrapper;
  beforeEach(() => {
    onSubmit = jest.fn();
    history = {
      push: jest.fn()
    }
    wrapper = shallow(<CreateExpensePage addExpense={onSubmit} history={history}/>);
  })

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should handle onSubmit', () => {
    const expense = {
      id: '3',
      desc: 'Rent', 
      note: 'The rent was not cheap either', 
      amount: '2000000', 
      createdAt: moment(0)
    };
     wrapper.find('ExpenseForm').prop('onSubmit')(expense);
     expect(history.push).toHaveBeenLastCalledWith('/');
     expect(onSubmit).toHaveBeenLastCalledWith(expense);
  })
})