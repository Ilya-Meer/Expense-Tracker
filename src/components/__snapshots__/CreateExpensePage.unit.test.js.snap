// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Expense Page should render properly 1`] = `
<div>
  <styled.h2>
    Add an Expense
  </styled.h2>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
